src/Microsoft.Dnx.Tooling/Restore/ImplicitPackagesWalkProvider.cs

```diff
@@ -33,14 +47,30 @@ internal class ImplicitPackagesWalkProvider : IWalkProvider
             new RuntimeSpec("win10-x86", "win10", "win81-x86"),
             new RuntimeSpec("win10-x64", "win10", "win81-x64"),
             new RuntimeSpec("win10-arm", "win10", "win81-arm"),
+
             new RuntimeSpec("unix", "any"),
-            new RuntimeSpec("osx.10.10", "unix"),
-            new RuntimeSpec("osx.10.10-x64", "osx.10.10"),
+            new RuntimeSpec("unix-x64", "unix"),
+
+            new RuntimeSpec("osx", "unix"),
+            new RuntimeSpec("osx-x64", "osx", "unix-x64"),
+
+            new RuntimeSpec("osx.10.10", "osx"),
+            new RuntimeSpec("osx.10.10-x64", "osx.10.10", "osx-x64"),
+
             new RuntimeSpec("linux", "unix"),
-            new RuntimeSpec("centos.7.1", "linux"),
-            new RuntimeSpec("centos.7.1-x64", "centos.7.1"),
-            new RuntimeSpec("ubuntu.14.04", "linux"),
-            new RuntimeSpec("ubuntu.14.04-x64", "ubuntu.14.04"));
+            new RuntimeSpec("linux-x64", "linux", "unix-x64"),
+
+            new RuntimeSpec("centos", "linux"),
+            new RuntimeSpec("centos-x64", "centos", "linux-x64"),
+
+            new RuntimeSpec("centos.7.1", "centos"),
+            new RuntimeSpec("centos.7.1-x64", "centos.7.1", "centos-x64"),
+
+            new RuntimeSpec("ubuntu", "linux"),
+            new RuntimeSpec("ubuntu-x64", "ubuntu", "linux-x64"),
+
+            new RuntimeSpec("ubuntu.14.04", "ubuntu"),
+            new RuntimeSpec("ubuntu.14.04-x64", "ubuntu.14.04", "ubuntu-x64"));
```

Comment:
I don't think so. It would take more effort than it's worth since we only need to modify this when DNX adds support for a new platform.